model_compression_toolkit/pytorch/back2framework/model_builder.py

```diff
@@ -0,0 +1,196 @@
+                             inputs: Tuple[Any],
+                             node_to_output_tensors_dict: Dict[BaseNode, List]) -> List[List]:
+    """
+    Given a node, build a list of input tensors the node gets. The list is built
+    based on the node's incoming edges and previous nodes' output tensors.
+
+    Args:
+        node: Node to build its input tensors list.
+        graph: Graph the node is in.
+        inputs: list of input tensors to model
+        node_to_output_tensors_dict: A dictionary from a node to its output tensors.
+
+    Returns:
+        A list of the node's input tensors.
+    """
+    if node.layer_class == DummyPlaceHolder:
+        input_tensors = inputs[graph.get_inputs().index(node)]
+    else:
+        input_tensors = []
+        # Go over a sorted list of the node's incoming edges, and for each source node get its output tensors.
+        # Append them in a result list.
+        for ie in graph.incoming_edges(node, sort_by_attr=EDGE_SINK_INDEX):
+            _input_tensors = node_to_output_tensors_dict[ie.source_node]
+            input_tensors.append(_input_tensors)
+    return input_tensors
+
+
+def run_operation(n: BaseNode,
+                  input_tensors: List,
+                  op_func: Any,
+                  mode: ModelBuilderMode = ModelBuilderMode.QUANTIZED) -> List[torch.Tensor]:
+    """
+    Applying the layer (op_func) to the input tensors (input_tensors).
+    If quantized is set to True, and the layer's corresponding node (n) has quantization
+    attributes, an additional fake-quantization node is built and appended to the layer.
+
+    Args:
+        n: The corresponding node of the layer it runs.
+        input_tensors: List of Pytorch tensors that are the layer's inputs.
+        op_func: Module/functional to apply to the input tensors.
+        mode: model quantiztion mode from ModelBuilderMode
+
+    Returns:
+        A list of Pytorch tensors. The Module/functional output tensors after applying the
+        Module/functional to the input tensors.
+    """
+
+    if n.layer_class == DummyPlaceHolder:  # Placeholder handling
+        out_tensors_of_n = input_tensors
+    else:
+        input_tensors = [tensor for tensor_list in input_tensors for tensor in tensor_list]  # flat list of lists
+        op_call_args = n.op_call_args if isinstance(n, FunctionalNode) else []
+        functional_kwargs = n.op_call_kwargs if isinstance(n, FunctionalNode) else {}
+        if isinstance(n, FunctionalNode) and n.inputs_as_list:
+            out_tensors_of_n = op_func(input_tensors, *op_call_args, **functional_kwargs)
+        else:
+            out_tensors_of_n = op_func(*input_tensors + op_call_args, **functional_kwargs)
+
+    # Add a fake quant node if the node has an activation threshold.
+    if mode == ModelBuilderMode.QUANTIZED and n.is_activation_quantization_enabled():
+        fake_quant = n.activation_quantization_cfg.generate_quantization_node()
```

Comment:
I think this code (lines 94-98) repeats itself in Keras; let's make it a function in common?